coderd/files/cache.go

```diff
@@ -163,11 +180,20 @@ func (c *Cache) Acquire(ctx context.Context, fileID uuid.UUID) (fs.FS, error) {
 	}
 	// Always check the caller can actually read the file.
 	if err := c.authz.Authorize(ctx, subject, policy.ActionRead, it.Object); err != nil {
-		c.Release(fileID)
+		c.release(fileID)
 		return nil, err
 	}
 
-	return it.FS, err
+	var once sync.Once
```

Comment:
I'd rather store the `Once` on the `struct` and then just put the closing logic directly in the `Close` method.

Reply:
You also have to store the `files.Cache` on the struct then too. 
Which allows calling `Acquire` and other methods.

It would be something like:
```go
type CloseFS struct {
	fs.FS
	
	once sync.Once
	fileID uuid.UUID
	cache *Cache
}

func (f *CloseFS) Close() error {
	f.once.Do(func() {
		f.cache.release(f.fileID)
	})
}
```

I do not want `cache` to be accessible, so I could throw the `release` as a method, but then we're back to having an anonymous function as a field. In that case, I'd rather not add fields to the struct that are only used in `Close`